pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/impl/antlr4/AntlrToken.java

```diff
@@ -4,42 +4,37 @@
 
 package net.sourceforge.pmd.lang.ast.impl.antlr4;
 
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-
 import org.antlr.v4.runtime.Lexer;
 import org.antlr.v4.runtime.Token;
 
 import net.sourceforge.pmd.annotation.Experimental;
 import net.sourceforge.pmd.lang.ast.GenericToken;
+import net.sourceforge.pmd.lang.document.FileLocation;
+import net.sourceforge.pmd.lang.document.TextDocument;
+import net.sourceforge.pmd.lang.document.TextRegion;
 
 /**
  * Generic Antlr representation of a token.
  */
 public class AntlrToken implements GenericToken<AntlrToken> {
 
-    private static final Pattern NEWLINE_PATTERN =
-        // \R on java 8+
-        Pattern.compile("\\u000D\\u000A|[\\u000A\\u000B\\u000C\\u000D\\u0085\\u2028\\u2029]");
-
```

Comment:
This looked much more complete than just `\r\n|\n`, which is used in TextFileContent...

Reply:
TextFileContent uses `\r\n?|\n`, so it handles three line terms. This is consistent with many Java core APIs like [`String#lines()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#lines()) or [`BufferedReader::readLine`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/BufferedReader.html#readLine())

Reply:
:+1: Yes, I think it's good to be consistent.